web/packages/teleterm/src/ui/ConnectMyComputer/connectMyComputerContext.tsx

```diff
@@ -207,6 +223,40 @@ export const ConnectMyComputerContextProvider: FC<{
     }, [connectMyComputerService, rootClusterUri, workspacesService])
   );
 
+  const [removeAgentAttempt, removeAgent] = useAsync(
+    useCallback(async () => {
+      const [, error] = await killAgent();
+      if (error) {
+        throw error;
+      }
+      setCurrentActionKind('remove');
+
+      try {
+        await retryWithRelogin(ctx, rootClusterUri, () =>
+          ctx.connectMyComputerService.removeConnectMyComputerNode(
+            rootClusterUri
+          )
+        );
+      } catch (e) {
+        if (isAccessDeniedError(e)) {
+          ctx.notificationsService.notifyInfo({
+            title: 'The node may be visible for a few more minutes.',
+            description:
+              'You do not have permissions to remove nodes, but it will be removed automatically.',
+          });
+        } else {
+          throw e;
+        }
+      }
+
+      await ctx.connectMyComputerService.removeConnections(rootClusterUri);
```

Comment:
Let's document that the order is important because we get the node UUID from the agent directory.